### DISCUSSION TITLE
I started using postgres with next js 14/13 and initially it worked just fine, but out of no where there is compile error with the following error. Can anyone guide on what should be done?

### DISCUSSION THREAD

**Discussion by codeAdityagithub** [porsager/postgres#755] [Q&A]
Module build failed: UnhandledSchemeError: Reading from "cloudflare:sockets" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "cloudflare:" URIs.
Import trace for requested module:
cloudflare:sockets
./node_modules/postgres/cf/polyfills.js
./node_modules/postgres/cf/src/index.js
./src/utils/db.ts
./src/app/api/auth/[...nextauth]/route.ts
![Screenshot 2023-12-11 185236](https://github.com/porsager/postgres/assets/89391179/fd59c5fa-061e-4289-b83a-ee56ae085f4b)

**Comment by kam-st** [comment]
same issues, were you able to resolve this.

**Comment by porsager** [comment]
I think upgrading to 3.4.4 should do it

**Reply by kam-st** [reply]
Using 3.4.4 still fiving following error in nextjs.

```
⨯ cloudflare:sockets
Module build failed: UnhandledSchemeError: Reading from "cloudflare:sockets" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "cloudflare:" URIs.
Import trace for requested module:
cloudflare:sockets
./node_modules/postgres/cf/polyfills.js
./node_modules/postgres/cf/src/index.js
```

**Comment by porsager** [comment]
Sorry. My mistake. Forgot that I hadn't released the latest on main.. Try `npm I porsager/postgres`On 7 Jun 2024, at 23.22, Kamal Singh ***@***.***> wrote:﻿
Using 3.4.4 still fiving following error in nextjs.
⨯ cloudflare:sockets
Module build failed: UnhandledSchemeError: Reading from "cloudflare:sockets" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "cloudflare:" URIs.
Import trace for requested module:
cloudflare:sockets
./node_modules/postgres/cf/polyfills.js
./node_modules/postgres/cf/src/index.js

**Reply by kam-st** [reply]
Thank you so much. Resolves cloudflare:sockets error.

**Reply by Harsh2509** [reply]
Getting this:-

bun add porsager/postgres
[1.27ms] ".env"
bun add v1.1.26 (0a37423b)
error: Package "postgres@3.4.4" has a dependency loop
  Resolution: "postgres@github:porsager/postgres#1818699"
  Dependency: "postgres@^3.4.4"
error: An internal error occurred (DependencyLoop)

**Reply by byte47** [reply]
@porsager is there any reason this fix is not deployed to npm?. 

I tried the above fix you mentioned , it was working for a while, then started showing some other error 

```
Module not found: Can't resolve 'net'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/.pnpm/postgres@https+++codeload.github.com+porsager+postgres+tar.gz+18186998b89a8ec60b82fd7140783f8833810e2d/node_modules/postgres/src/index.js
./src/db/index.ts
./src/auth.ts
./src/app/(auth)/_components/user-auth-form.tsx
```
